docs/analytics-ga4/ga4-ecommerce-tracker.md

```diff
@@ -17,22 +17,24 @@ In `analytics-ga4.js`:
 In `live_search.js` in the `finder-frontend` repository (https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L129):
 
 ```JavaScript
-GOVUK.analyticsGa4.Ga4EcommerceTracker.init(isNewPage) // isNewPage is a boolean and is used to detect whether a new page has been loaded or not
+GOVUK.analyticsGa4.Ga4EcommerceTracker.init(dynamicPageViewData) 
+
+/* The init() function takes an object (dynamicPageViewData) as an optional parameter to detect whether dynamic updates have been made to the page via AJAX requests. The values in the object are then used to determine what is pushed to the dataLayer. If a parameter is not passed, this indicates a fresh page load (e.g. by refreshing the page or using the pagination) and that the page has not been dynamically updated via JS. */
 ```
 
 Unlike the other GA4 tracking scripts in this repository, ecommerce tracking is not initialised by `init-ga4.js`. Instead, it is initialised by the JS that runs on finder pages (specifically, the `LiveSearch.prototype.Ga4EcommerceTracking()` function). It is called:
 
-On page load - https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L55
+On page load (where no parameters are passed) - https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L55
 
-And whenever search results are updated - https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L121
+And whenever search results are updated (where an object parameter is passed) - https://github.com/alphagov/finder-frontend/blob/main/app/assets/javascripts/live_search.js#L121
```

Comment:
There's a few links to specific line numbers in github files in this doc, which seems like it'll get out of date quickly (there's a way of making a more permanent link although I can't remember off the top of my head what it is).

As a general rule it would be good if this documentation was standalone even if e.g. github was down. Are these links simply references to 'this is where this happens in the code'? Is there any more explanation you could add here to mean that readers don't always have to follow the links?

Reply:
Very true. I’ve removed the links now and just put code snippets in as they were just there to show what happens in the code, as you say